lib/absinthe/phase/document/validation/no_fragment_cycles.ex

```diff
@@ -35,7 +35,17 @@ defmodule Absinthe.Phase.Document.Validation.NoFragmentCycles do
   defp check(fragments) do
     graph = :digraph.new([:cyclic])
     try do
-      check(fragments, graph)
+      with {fragments, 0} <- check(fragments, graph) do
+        fragments = Map.new(fragments, &{&1.name, &1})
+
+        fragments =
+          graph
+          |> :digraph_utils.topsort
```

Comment:
This is in that phase, `topsort/1` is just a function that operates on the graph we already created to return a list of verticies.

I've considered keeping the `:digraph` table around for the rest of the request but it complicates resource management a bit. More to the point, once you've sorted the fragments list there's not much you need the graph for, since you've got a handy sorted tree list on hand already that doesn't require any fancy pointer / vertex tracking.